src/core/metrics.cc

```diff
@@ -365,7 +387,7 @@ Metrics::UUIDForCudaDevice(int cuda_device, std::string* uuid)
   }
 
   nvmlDevice_t device;
-  nvmlReturn_t nvmlerr = nvmlDeviceGetHandleByPciBusId(pcibusid_str, &device);
+  nvmlReturn_t nvmlerr = nvmlDeviceGetHandleByPciBusId_v2(pcibusid_str, &device);
   if (nvmlerr != NVML_SUCCESS) {
     LOG_ERROR << "failed to get device from PCI Bus ID: NVML_ERROR " << nvmlerr;
```

Comment:
Modify to `nvmlErrorString(nvmlerr)` instead

Reply:
Do you want me to change all the error handlings to this way? Since other API calls to nvml in this file handle the errors the same way.

/cc @deadeyegoodwin

Reply:
Yes, it would be a good improvement to update all the error handling to give a better message.